src/main/java/uk/gov/ons/ctp/response/action/service/ActionProcessingService.java

```diff
@@ -35,66 +33,119 @@
   public static final String CREATED = "CREATED";
   public static final String ENABLED = "ENABLED";
   public static final String NOTIFY = "Notify";
+  public static final String PRINTER = "Printer";
   public static final String PENDING = "PENDING";
 
-  @Autowired private ActionRepository actionRepo;
-
-  @Autowired private ActionInstructionPublisher actionInstructionPublisher;
+  @Autowired private ActionCaseRepository actionCaseRepo;
 
   @Autowired private NotifyService notifyService;
 
-  @Autowired
-  private StateTransitionManager<ActionDTO.ActionState, ActionDTO.ActionEvent>
-      actionSvcStateTransitionManager;
+  @Autowired private NotificationFileCreator notificationFileCreator;
+
+  @Autowired private ActionStateService actionStateService;
 
-  private ActionRequestDecorator[] decorators;
+  @Autowired private ActionRequestContextFactory decoratorContextFactory;
 
-  public abstract ActionRequestContextFactory getActionRequestDecoratorContextFactory();
+  private static final ActionRequestDecorator[] DECORATORS = {
+    new ActionAndActionPlan(),
+    new CaseAndCaseEvent(),
+    new CollectionExerciseAndSurvey(),
+    new PartyAndContact(),
+    new SampleUnitRef()
+  };
 
-  public ActionProcessingService(ActionRequestDecorator[] decorators) {
-    this.decorators = decorators;
+  public void processActions(ActionType actionType, List<Action> allActions) {
+    if (actionType == null) {
+      throw new IllegalStateException(ACTION_TYPE_NOT_DEFINED);
+    }
+    String handler = actionType.getHandler();
+    if (handler == null) {
+      log.with("handler", actionType.getHandler()).error("no action type handler provided");
```

Comment:
Should probably log the name of the actionType so we can know which one doesn't have a handler

Reply:
done 👍